### PR TITLE
Honor `AccessibleTopology` during `Volume` creation

### PR DESCRIPTION
**Pull request by FlorinPeter**
# Proposed Changes

- fix for missing accessible topology handling

### CODE REVIEW

File: pkg/driver/driver_test.go [hunk 1]
```diff
@@ -34,17 +34,20 @@ var _ = Describe("Service tests", func() {
 	var (
 		reqVolumeId    = "v1"
 		testDeviceName = "sda"
+		reqVolumePool  = "pool1"
```
Comment (afritzler) [inline pkg/driver/driver_test.go:37]: Please don't prefix variables with `req` or `request`.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: pkg/driver/controller.go [hunk 2]
```diff
@@ -52,9 +52,17 @@ func (d *driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
...
+	var accessibleTopology []*csi.Topology
 	// if no volume_pool was provided try to use the topology information if provided
 	if volumePool == "" && req.GetAccessibilityRequirements() != nil {
 		volumePool = getAZFromTopology(req.GetAccessibilityRequirements())
+		accessibleTopology = []*csi.Topology{
+			{
+				Segments: map[string]string{topologyKey: volumePool},
+			},
+		}
+	} else if volumePool == "" && req.GetAccessibilityRequirements() == nil {
```
Comment (afritzler) [inline pkg/driver/controller.go:64]: Please use this as an `if` statement and return early the error.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: pkg/driver/driver_test.go (hunk 1 above)
Comment (sujeet01) [reply pkg/driver/driver_test.go]: Done

File: pkg/driver/controller.go (hunk 2 above)
Comment (sujeet01) [reply pkg/driver/controller.go]: Done

**Review by afritzler — APPROVED** [review]

**Pull request merged**